### PR TITLE
[MLv2] Implement `drillThru()` for pivot drills; add BE unit tests

### LINKED ISSUE #33559: [MLv2] [BE] Add `drill-thru/pivot` aka `Break out by...`
See @bshepherdson 's brain dump thread for more info https://metaboat.slack.com/archives/C04CYTEL9N2/p1691679718619139

Specifically

> pivoting returns a big map of possible columns to pivot on, grouped by the kind of column (category, time, location). the FE needs this to produce the 2-click nested view for that particular drill (try it!)

I think we need to fill out the `drill-thru-method` for pivots, I **think** one of these files has the FE code we need to port but don't quote me on that.

https://github.com/metabase/metabase/blob/0b0edf1b916aa6a4000767ec42a41c2721d22391/frontend/src/metabase-lib/queries/utils/pivot.js#L4

https://github.com/metabase/metabase/blob/0b0edf1b916aa6a4000767ec42a41c2721d22391/frontend/src/metabase-lib/queries/drills/pivot-drill.ts

1. Port JS code to Cljc
2. Fill out the skeleton `drill-thru-method` 
3. Make sure it comes back when expected in `available-drill-thrus`
4. Write a billion tests

### PR DESCRIPTION
**Pull request by bshepherdson**
Fixes #33559.

### CODE REVIEW

**Comment by bshepherdson** [comment]
Current dependencies on/for this PR:
* master
  * **PR #36281** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36281?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/36281?utm_source=stack-comment).

**Review by ranquild — APPROVED** [review]

**Review by metamben — APPROVED** [review]

File: src/metabase/lib/drill_thru/pivot.cljc [hunk 1]
```diff
@@ -55,47 +56,68 @@
...
+    (let [breakout-pivot-types (permitted-pivot-types query stage-number)
+          by-category          (when (breakout-pivot-types :category)
+                                 (pivot-by-category-drill query stage-number context))
+          by-location          (when (breakout-pivot-types :location)
+                                 (pivot-by-location-drill query stage-number context))
+          by-time              (when (breakout-pivot-types :time)
+                                 (pivot-by-time-drill     query stage-number context))
+          pivots               (merge (when (seq by-category) {:category by-category})
+                                      (when (seq by-location) {:location by-location})
+                                      (when (seq by-time)     {:time by-time}))]
```
[the remark this answers is not in the document]
Comment (bshepherdson) [reply src/metabase/lib/drill_thru/pivot.cljc]: I wasn't thrilled with how that looked either, but I think I found a winning factoring using `(into {} (for ...))`.

File: src/metabase/lib/drill_thru/pivot.cljc [hunk 2]
```diff
@@ -55,47 +39,64 @@
...
-          pivots      (merge (when (seq by-category) {:category by-category})
-                             (when (seq by-location) {:location by-location})
-                             (when (seq by-time)     {:time by-time}))]
-      ;; TODO: Do dimensions need to be attached? How is clicked.dimensions calculated in the FE?
+    (let [breakout-pivot-types (permitted-pivot-types query stage-number)
+          pivots               (into {} (for [pivot-type breakout-pivot-types
+                                              :let [pred    (get pivot-type-predicates pivot-type)
+                                                    columns (pivot-drill-pred query stage-number context pred)]
+                                              :when (not-empty columns)]
+                                          [pivot-type columns]))]
```
Comment (metamben) [inline src/metabase/lib/drill_thru/pivot.cljc:94]: This is much better, indeed!

File: src/metabase/lib/drill_thru/pivot.cljc [hunk 3]
```diff
@@ -55,47 +39,64 @@
...
-                        (pivot-by-time-drill     query stage-number context))
-          pivots      (merge (when (seq by-category) {:category by-category})
-                             (when (seq by-location) {:location by-location})
-                             (when (seq by-time)     {:time by-time}))]
-      ;; TODO: Do dimensions need to be attached? How is clicked.dimensions calculated in the FE?
+    (let [breakout-pivot-types (permitted-pivot-types query stage-number)
+          pivots               (into {} (for [pivot-type breakout-pivot-types
+                                              :let [pred    (get pivot-type-predicates pivot-type)
+                                                    columns (pivot-drill-pred query stage-number context pred)]
+                                              :when (not-empty columns)]
```
Comment (metamben) [inline src/metabase/lib/drill_thru/pivot.cljc:93]: Could have been `(seq columns)`.

**Pull request merged**